editor/src/components/canvas/canvas-strategies/relative-move-strategy.tsx

```diff
@@ -0,0 +1,188 @@
+
+    return hasOffsets
+      ? 4 // +1 than reorder flow
+      : 1 // there should be a more structured way to define priorities (:
+  },
+
+  apply: (canvasState, interactionState, sessionState) => {
+    const { interactionData } = interactionState
+    if (!(interactionData.type === 'DRAG' && interactionData.drag != null)) {
+      return emptyStrategyApplicationResult
+    }
+
+    const selectedElements = getTargetPathsFromInteractionTarget(canvasState.interactionTarget)
+    if (selectedElements.length === 0) {
+      return emptyStrategyApplicationResult
+    }
+
+    const filteredSelectedElements = getDragTargets(selectedElements)
+    const last = filteredSelectedElements[filteredSelectedElements.length - 1]
+    const meta = MetadataUtils.findElementByElementPath(sessionState.startingMetadata, last)
+    if (meta == null) {
+      return emptyStrategyApplicationResult
+    }
+    const offsets = getStyleOffsets(meta)
+    if (!offsets) {
+      return emptyStrategyApplicationResult
+    }
+
+    const { drag } = interactionData
+
+    const commands: Array<CanvasCommand> = []
+
+    // should set a direction only if it's either defined in the original offsets or if the counterpart (horiz/vert) is not defined
+    // vertical
+    commands.push(
+      ...applyStyle({
+        name: 'top',
+        initial: offsets.top,
+        delta: drag.y,
+        keep: offsets.bottom == null,
+        path: last,
+      }),
+      ...applyStyle({
+        name: 'bottom',
+        initial: offsets.bottom,
+        delta: -drag.y,
+        path: last,
+      }),
+    )
+    // horizontal
+    commands.push(
+      ...applyStyle({
+        name: 'left',
+        initial: offsets.left,
+        delta: drag.x,
+        keep: offsets.right == null,
+        path: last,
+      }),
+      ...applyStyle({
+        name: 'right',
+        initial: offsets.right,
+        delta: -drag.x,
+        path: last,
+      }),
+    )
+
+    return strategyApplicationResult(commands)
+  },
+}
+
+const applyStyle = (params: {
+  name: 'top' | 'left' | 'bottom' | 'right'
+  initial: number | null
+  delta: number
+  keep?: boolean // if true, this item will be preferred
+  path: ElementPath
+}): CanvasCommand[] => {
+  const { name, initial, delta, keep, path } = params
+
+  const skip = initial == null && !keep
+  if (skip) {
+    return []
+  }
+
+  const value = (initial || 0) + delta
+  return [
+    setCssLengthProperty('always', path, stylePropPathMappingFn(name, ['style']), value, undefined),
+  ]
+}
+
+const getStyleOffsets = (meta: ElementInstanceMetadata) => {
+  const getOffsetPropValue = (
+    name: 'left' | 'top' | 'right' | 'bottom',
+    attrs: PropsOrJSXAttributes,
+  ): number | null => {
+    const prop = getLayoutProperty(name, attrs, ['style'])
+    if (!isRight(prop)) {
+      return null
+    }
+    if (!prop.value) {
+      return null
+    }
+    return prop.value.value
```

Comment:
You might benefit from using [`foldEither`](https://github.com/concrete-utopia/utopia/blob/master/editor/src/core/shared/either.ts#L41-L51) here (or some of the other functions in that file)